compiler/src/dotty/tools/dotc/typer/FrontEnd.scala

```diff
@@ -80,6 +80,8 @@ class FrontEnd extends Phase {
         typr.println("typed: " + unit.source)
         record("retained untyped trees", unit.untpdTree.treeSize)
         record("retained typed trees after typer", unit.tpdTree.treeSize)
+        if unit.isJava then
+          JavaChecks.check(unit.tpdTree)
```

Comment:
I think we should run these checks in `runOn` on all Java units at once after `unitContexts.foreach(typeCheck(using _))` for the same reason we normally run them in PostTyper: doing these checks in the middle of typechecking could lead to cycles

Reply:
done!